cmd/gapit/common.go

```diff
@@ -111,8 +113,39 @@ func getDevice(ctx context.Context, client client.Client, capture *path.Capture,
 		return nil, log.Err(ctx, err, "Failed query list of devices for replay")
 	}
 
-	if len(paths) > 0 {
-		return paths[0], nil
+	filteredByFlags := make([]*path.Device, 0, len(paths))
+	for i := 0; i < len(paths); i++ {
+		p := paths[i]
+		o, err := client.Get(ctx, p.Path())
+		if err != nil {
+			return nil, log.Err(ctx, err, "Couldn't resolve device")
+		}
+		d := o.(*device.Instance)
+		switch flags.Device {
+		case "":
+			// empty flag
+			filteredByFlags = append(filteredByFlags, p)
+		case "host":
+			if d == host.Instance(ctx) {
+				filteredByFlags = append(filteredByFlags, p)
+				break
+			}
+		case "android":
+			if d.GetConfiguration().GetOS().GetKind() == device.OSKind_Android {
+				filteredByFlags = append(filteredByFlags, p)
+			}
+		default:
+			// serial number
+			// TODO: Regex matching instead of exact match?
+			if d.GetSerial() == flags.Device {
+				filteredByFlags = append(filteredByFlags, p)
+				break
+			}
+		}
+	}
+
+	if len(filteredByFlags) > 0 {
+		return filteredByFlags[0], nil
```

Comment:
In the case of `> 1`, perhaps print there were multiple usable devices, and which one was picked.